src/web/ViewCubeGizmo.tsx

```diff
@@ -0,0 +1,138 @@
+import * as React from 'react'
+import { useGizmoContext } from '../core/GizmoHelper'
+import { CanvasTexture, Event, Vector3 } from 'three'
+
+type XYZ = [number, number, number]
+
+const faces = ['right', 'left', 'top', 'bottom', 'front', 'back']
+
+const makePositionVector = (xyz) => new Vector3(...xyz).multiplyScalar(0.38)
+
+const corners: Vector3[] = [
+  [1, 1, 1],
+  [1, 1, -1],
+  [1, -1, 1],
+  [1, -1, -1],
+  [-1, 1, 1],
+  [-1, 1, -1],
+  [-1, -1, 1],
+  [-1, -1, -1],
+].map(makePositionVector)
+
+const cornerDimensions: XYZ = [0.25, 0.25, 0.25]
+
+const edges: Vector3[] = [
+  [1, 1, 0],
+  [1, 0, 1],
+  [1, 0, -1],
+  [1, -1, 0],
+  [0, 1, 1],
+  [0, 1, -1],
+  [0, -1, 1],
+  [0, -1, -1],
+  [-1, 1, 0],
+  [-1, 0, 1],
+  [-1, 0, -1],
+  [-1, -1, 0],
+].map(makePositionVector)
+
+const edgeDimensions = edges.map(
+  (edge) => edge.toArray().map((axis: number): number => (axis == 0 ? 0.5 : 0.25)) as XYZ
+)
+
+const FaceMaterial = ({ hover, index }: { hover: boolean; index: number }) => {
+  const texture = React.useMemo(() => {
+    const canvas = document.createElement('canvas')
+    canvas.width = 128
+    canvas.height = 128
+
+    const context = canvas.getContext('2d')!
+    context.fillStyle = '#eee'
+    context.fillRect(0, 0, canvas.width, canvas.height)
+    context.strokeStyle = '#444'
+    context.strokeRect(0, 0, canvas.width, canvas.height)
+
+    context.font = '28px Arial'
+    context.textAlign = 'center'
+    context.fillStyle = '#222'
+    context.fillText(faces[index].toUpperCase(), 64, 76)
+    return new CanvasTexture(canvas)
+  }, [index])
+  return <meshLambertMaterial map={texture} attachArray="material" color={hover ? 'hotpink' : 'white'} />
+}
+
+const FaceCube = () => {
+  const { tweenCamera, raycast } = useGizmoContext()
+  const [hover, set] = React.useState<number | null>(null)
+  const handlePointerDown = (e: Event) => {
+    tweenCamera(e.face.normal)
+    e.stopPropagation()
+  }
+  const handlePointerMove = (e: Event) => {
+    set(Math.floor(e.faceIndex / 2))
+    e.stopPropagation()
+  }
+  return (
+    <mesh
+      raycast={raycast}
+      onPointerOut={() => set(null)}
```

Comment:
I know it's one line, but for consistency, let's move it to it's own func `handlePointerOut`